mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -159,6 +159,8 @@ void AffixTuningParameters::affixTuningParametersImpl(T &op) {
     op->setAttr("n_per_wave", b.getI32IntegerAttr(validParams.gemmNPerWave));
     op->setAttr("block_size", b.getI32IntegerAttr(blockSize));
 
+    // Set attributes on the function.
+    getFunction()->setAttr("kernel", b.getI32IntegerAttr(1));
```

Comment:
If this is always 1, why do we need this attribute? I don't see other behavior when you do the `MIOpenToGpu` lowering when it is not 1.

Reply:
`1` is just a placeholder as we need a key/value pair in the function attribute.

Reply:
Would there be a case in the future that `getAttr("kernel")` for some `FuncOp` would not be 1?

Reply:
I just updated the implementation so we use `UnitAttr` now. There would be no `1` anymore. Also added a unit test.

Reply:
Looking good. This make sure one funcOp maps to exactly 1 kernel; No more, no less.